### ISSUE TITLE
Performance counters do not compile

### ISSUE THREAD

**Issue by diehlpk** [STEllAR-GROUP/hpx#5856]
## Expected Behavior

HPX master should compile with armclang on A64fx.

## Actual Behavior

```

/lustre/projects/diehl-group/nan/OctoTigerBuildChain-clang/src/hpx/libs/full/performance_counters/src/threadmanager_counter_types.cpp:476:36: error: use of undeclared identifier 'counter_average_count'
            {"/threads/idle-rate", counter_average_count,
                                   ^
/lustre/projects/diehl-group/nan/OctoTigerBuildChain-clang/src/hpx/libs/full/performance_counters/src/threadmanager_counter_types.cpp:787:34: error: invalid application of 'sizeof' to an incomplete type 'hpx::performance_counters::generic_counter_type_data []'
            counter_types, sizeof(counter_types) / sizeof(counter_types[0]));
                                 ^~~~~~~~~~~~~~~
2 errors generated.
```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. Compile HPX master with armclang

## Specifications

... Please describe your environment

  - HPX Version: master
  - Platform (compiler, OS): armclang++

**Comment by hkaiser** [comment]
This will be fixed by #5854

**Comment by gonidelis** [comment]
Apologies for that Patrick! I modernized certain names and I missed a couple. #5854 should be ready by tomorrow.
